### PR TITLE
Add Enum key support for TabNavigation

### PR DESCRIPTION
**Pull request by sleushunou**
### Bootstrapper

- Skip auto-register abstract ViewModels;

### BottomTabNavigation
- Changed `TabItem` to `TabItem<TKey>`
- `TabItem` needs `IContainer` explicitly via ctor
- Changed `ToolbarViewModelBase` to ToolbarViewModelBase<TKey>
- iOS:
  - Changed `ToolbarViewControllerBase<TViewModel>` to `ToolbarViewControllerBase<TViewModel, TKey>`
- Android:
  - Changed `ToolbarActivityBase<TViewModel>` to `ToolbarActivityBase<TViewModel, TKey>`
- Added ability to getting custom icon by item key, via override `GetImageResourceId`

### CODE REVIEW

File: Softeq.XToolkit.WhiteLabel.iOS/IosBootstraper.cs [hunk 1]
```diff
@@ -52,9 +52,11 @@ protected override void RegisterInternalServices(IContainerBuilder builder)
             {
                 var viewModelType = type.BaseType.GetGenericArguments()[0];
 
-                viewModelToViewControllerTypes.Add(viewModelType, type);
-
-                builder.PerDependency(viewModelType, IfRegistered.Keep);
+                if (!viewModelType.IsGenericType)
```
[the remark this answers is not in the document]
Comment (sleushunou) [reply Softeq.XToolkit.WhiteLabel.iOS/IosBootstraper.cs]: Changed, now we skip abstract view models

File: Softeq.XToolkit.WhiteLabel/Model/TabItem.cs [hunk 2]
```diff
@@ -1,37 +1,41 @@
...
         }
 
-        public override TabViewModel CreateViewModel()
+        protected IContainer Container { get; }
+
+        public override TabViewModel<TKey> CreateViewModel()
         {
-            var frameNavigationService = Dependencies.Container.Resolve<IFrameNavigationService>();
-            var tabViewModel = new TabViewModel<TFirstViewModel>(frameNavigationService);
+            var frameNavigationService = Container.Resolve<IFrameNavigationService>();
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel/Model/TabItem.cs]: As idea: create a separate abstraction for getting `IFrameNavigationService`.

We should avoid using `IContainer` explicitly in high-level classes.

In this case, I prefer to keep using `Dependencies`.

File: Softeq.XToolkit.WhiteLabel/ViewModels/Tab/TabViewModel.cs [hunk 3]
```diff
@@ -30,9 +31,29 @@ public bool IsBadgeVisible
...
+        }
 
-        public string ImageKey => _tab.ImageKey;
+        public TKey Key
+        {
+            get
+            {
+                if (_tab == null)
+                {
+                    throw new Exception("ViewModel is not initialized");
```
[the remark this answers is not in the document]
Comment (wcoder) [reply Softeq.XToolkit.WhiteLabel/ViewModels/Tab/TabViewModel.cs]: Fixed

File: Softeq.XToolkit.WhiteLabel/ViewModels/Tab/TabViewModel.cs [hunk 4]
```diff
@@ -30,9 +31,29 @@ public bool IsBadgeVisible
...
         }
 
-        public string Title => _tab.Title;
+        public string Title
+        {
+            get
+            {
+                if (_tab == null)
+                {
+                    throw new Exception("ViewModel is not initialized");
```
[the remark this answers is not in the document]
Comment (wcoder) [reply Softeq.XToolkit.WhiteLabel/ViewModels/Tab/TabViewModel.cs]: Fixed

File: Softeq.XToolkit.WhiteLabel.iOS/IosBootstraper.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (wcoder) [reply Softeq.XToolkit.WhiteLabel.iOS/IosBootstraper.cs]: Fixed

File: Softeq.XToolkit.WhiteLabel/ViewModels/Tab/TabViewModel.cs [hunk 5]
```diff
@@ -41,13 +62,13 @@ public void GoBack()
...
 
-        internal void Initialize(TabItem tab)
+        public void Initialize(TabItem<TKey> tab)
         {
             _tab = tab;
         }
     }
 
-    public class TabViewModel<T> : TabViewModel where T : ViewModelBase
+    public class TabViewModel<TFirstViewModel, TKey> : TabViewModel<TKey> where TFirstViewModel : ViewModelBase
```
[the remark this answers is not in the document]
Comment (wcoder) [reply Softeq.XToolkit.WhiteLabel/ViewModels/Tab/TabViewModel.cs]: Fixed

**Pull request merged**